### PR TITLE
Bugfix/cusparse mv alg default

### PR DESCRIPTION
**Pull request by vincentmr**
**Context:**
`conda-forge` CI fails because `CUSPARSE_MV_ALG_DEFAULT` is deprecated in favor of `CUSPARSE_SPMV_ALG_DEFAULT` and because of the requirements listed in `setup.py`.

**Description of the Change:**
Reduce requirements in `setup.py` and replace `CUSPARSE_MV_ALG_DEFAULT`.

**Benefits:**
Add PL-Lightning-GPU to Conda-Forge.

**Possible Drawbacks:**
`CUSPARSE_SPMV_ALG_DEFAULT` not defined on old version of CUDA, `CUSPARSE_MV_ALG_DEFAULT` deprecated since at least 11.4.

**Related GitHub Issues:**

### CODE REVIEW

File: pennylane_lightning_gpu/src/algorithms/ObservablesGPU.hpp [hunk 1]
```diff
@@ -562,7 +562,7 @@ class SparseHamiltonianGPU final : public ObservableGPU<T> {
             /* cusparseDnVecDescr_t */ sv_prime_descriptor,
             /* cudaDataType */ data_type,
             /* cusparseSpMVAlg_t */
-            CUSPARSE_MV_ALG_DEFAULT,
+            CUSPARSE_SPMV_ALG_DEFAULT,
```
Comment (mlxd) [inline pennylane_lightning_gpu/src/algorithms/ObservablesGPU.hpp:565]:
```suggestion
            CUSPARSE_SPMV_CSR_ALG1,
```

Let's try the above, as it mirrors what is at line 548.

**Review by mlxd — CHANGES_REQUESTED** [review]
Thanks @vincentmr 
For the failing test, let's try the explicit CSR alg1 call and see if that helps.

Also, as you say the support is more recently defined in the CUDA version. Right now we build with a 11.5 version minimum, but it would be god to know the threshold of support if we wish to adjust this number later.

**Comment by vincentmr** [comment]
> Also, as you say the support is more recently defined in the CUDA version. Right now we build with a 11.5 version minimum, but it would be good to know the threshold of support if we wish to adjust this number later.

`CUSPARSE_MV_ALG_DEFAULT` is deprecated as of [CUDA 11.2.1](https://docs.nvidia.com/cuda/archive/11.2.1/cusparse/index.html).

**Comment by multiphaseCFD** [comment]
> > Also, as you say the support is more recently defined in the CUDA version. Right now we build with a 11.5 version minimum, but it would be good to know the threshold of support if we wish to adjust this number later.
> 
> `CUSPARSE_MV_ALG_DEFAULT` is deprecated as of [CUDA 11.2.1](https://docs.nvidia.com/cuda/archive/11.2.1/cusparse/index.html).

> > Also, as you say the support is more recently defined in the CUDA version. Right now we build with a 11.5 version minimum, but it would be good to know the threshold of support if we wish to adjust this number later.
> 
> `CUSPARSE_MV_ALG_DEFAULT` is deprecated as of [CUDA 11.2.1](https://docs.nvidia.com/cuda/archive/11.2.1/cusparse/index.html).
--

<br class="Apple-interchange-newline">

 Just found something from nvidia docs. `CUSPARSE_SPMV_COO_ALG1
` Default algorithm for COO sparse matrix format. May produce slightly different results during different runs with the same input parameters. https://docs.nvidia.com/cuda/cusparse/

**Comment by vincentmr** [comment]
Thanks @multiphaseCFD, I could not reproduce the failure on Perlmutter and the results look quite similar so should we increase the tolerance? or use the slower version?

**Comment by multiphaseCFD** [comment]
> Thanks @multiphaseCFD, I could not reproduce the failure on Perlmutter and the results look quite similar so should we increase the tolerance? or use the slower version?

Do you mean it's A10g used in our CI tests that leads to failure?

**Comment by vincentmr** [comment]
> > Thanks @multiphaseCFD, I could not reproduce the failure on Perlmutter and the results look quite similar so should we increase the tolerance? or use the slower version?
> 
> Do you mean it's A10g used in our CI tests that leads to failure?

I just meant the test are not failing hard, but maybe just flaky. I was suggesting adjusting the tolerances in `allclose` asserts. For instance, `test_adjoint_SparseHamiltonian` is using the default but maybe `rtol=0, atol=1e-07` would be ok.

File: pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp [hunk 2]
```diff
@@ -1397,7 +1397,7 @@ class StateVectorCudaManaged
 
     /**
      * @brief Apply a given host or device-stored array representing the gate
-     * `matrix` to the statevector at qubit indices given by `tgts` and
+     * `matrix` to the state vector at qubit indices given by `tgts` and
```
Comment (mlxd) [inline pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp:1400]: Is this consistent with PennyLane and other Lightning packages use of the term?

Comment (vincentmr) [reply pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp:1400]: Good question. From what I can see, it's about half-half in the PLL-GPU source, but it was 2 vs 1 for `state vector` in `StateVectorCudaManaged.hpp`. In the PennyLane source code (e.g. docstrings) `state vector` seems favored. I could `sed` this one way or another if we want to pick a side for good :)

Comment (mlxd) [reply pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp:1400]: Sounds good. Let's opt for `state vector` then when discussing, as keep `statevector` for class or method specific.

**Review by mlxd — APPROVED** [review]
Thanks @vincentmr

**Pull request merged**